src/D2L.Bmx/OktaAuthenticator.cs

```diff
@@ -114,26 +119,142 @@ No config file found. Your Okta session will not be cached.
+		Uri orgUrl,
+		string user,
+		IOktaClientFactory oktaClientFactory,
+		bool nonInteractive,
+		bool experimentalBypassBrowserSecurity
+	) {
+		await using IBrowser? browser = await Browser.LaunchBrowserAsync( experimentalBypassBrowserSecurity );
+		if( browser is null ) {
+			return null;
+		}
+
+		if( !nonInteractive ) {
+			Console.Error.WriteLine( "Attempting to automatically login using Okta Desktop Single Sign-On." );
+		}
+		using var cancellationTokenSource = new CancellationTokenSource( TimeSpan.FromSeconds( 15 ) );
+		var sessionIdTcs = new TaskCompletionSource<string?>( TaskCreationOptions.RunContinuationsAsynchronously );
+		string? sessionId;
+
+		try {
+			using var page = await browser.NewPageAsync().WaitAsync( cancellationTokenSource.Token );
+			int attempt = 1;
+
+			page.Load += ( _, _ ) => _ = GetSessionCookieAsync();
+			await page.GoToAsync( orgUrl.AbsoluteUri ).WaitAsync( cancellationTokenSource.Token );
+			sessionId = await sessionIdTcs.Task;
+
+			async Task GetSessionCookieAsync() {
+				var url = new Uri( page.Url );
+				if( url.Host == orgUrl.Host ) {
+					string title = await page.GetTitleAsync().WaitAsync( cancellationTokenSource.Token );
+					// DSSO can sometimes takes more than one attempt.
+					// If the path is '/' with 'sign in' in the title, it means DSSO is not available and we should stop retrying.
+					if( title.Contains( "sign in", StringComparison.OrdinalIgnoreCase ) ) {
+						if( attempt < 3 && url.AbsolutePath != "/" ) {
+							attempt++;
+							await page.GoToAsync( orgUrl.AbsoluteUri ).WaitAsync( cancellationTokenSource.Token );
+						} else {
+							consoleWriter.WriteWarning(
+								"WARNING: Could not authenticate with Okta using Desktop Single Sign-On." );
+							sessionIdTcs.SetResult( null );
+						}
+						return;
+					}
+				}
+				var cookies = await page.GetCookiesAsync( orgUrl.AbsoluteUri ).WaitAsync( cancellationTokenSource.Token );
+				if( Array.Find( cookies, c => c.Name == "sid" )?.Value is string sid ) {
+					sessionIdTcs.SetResult( sid );
+				}
+			}
+		} catch( TaskCanceledException ) {
+			consoleWriter.WriteWarning( $"""
+				WARNING: Timed out when trying to create Okta session through Desktop Single Sign-On.
+				Check if the org '{orgUrl}' is correct. If running BMX with elevated privileges,
+				rerun the command with the '--experimental-bypass-browser-security' flag
+				"""
+			);
+			return null;
+		} catch( TargetClosedException ) {
+			consoleWriter.WriteWarning( """
+				WARNING: Failed to create Okta session through Desktop Single Sign-On as BMX is likely being run
+				with elevated privileges. Rerun the command with the '--experimental-bypass-browser-security' flag.
+				"""
+			);
+			return null;
+		} catch( Exception ) {
+			consoleWriter.WriteWarning(
+				"WARNING: Unknown error while trying to authenticate with Okta using Desktop Single Sign-On." );
+			return null;
+		}
+
+		if( sessionId is null ) {
+			return null;
+		}
```

Comment:
nit: all these `return null` in all the `catch` blocks seem unnecessary?
`if( sessionId is null ) { return null; }` should handle them all, right?